go/multipooler/grpcmanagerservice/service.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2025 Supabase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package grpcmanagerservice implements the gRPC server for MultiPoolerManager
+package grpcmanagerservice
```

Comment:
Introducing this pattern from Vitess. I think it makes it very obvious what is the gRPC server implementation and allows us to decouple the core business logic from the API.